### ISSUE TITLE
Thresholds for sensors

### ISSUE THREAD

**Issue by stephanemagnenat** [aseba-community/aseba#233]
In advanced mode, it would be nice to be able to alter the sensor thresholds.

**Comment by riedo** [comment]
I just noticed something while doing tests (btw, the "grey" state of the sensors helps a lot with the hands following, and reordering is not problematic anymore):

The sliders map values linearly between 0 and 4000, but in reality, when nothing is detected the value is 0, and as an object approaches it suddenly "jumps" to more than 1500; in addition, it often reaches more than 5000 if the hand comes really close. It depends on the sensors and thus it is not easy to find universal values values; however, in the current situation, only looking at VPL the position of the slider does not help much to predict at which distance the robot will react. 
For a hand following (just back and forth, without touching the hand) I could not do it with just VPL; I had to look at the text and auto-refreshed memory window to understand the problem. I add a screenshot underneath to explain the problem; it's not very intuitive looking IMO.

![screen shot 2015-08-10 at 3 07 00 pm](https://cloud.githubusercontent.com/assets/1862778/9172174/eef80364-3f71-11e5-8631-0924d5f447ad.png)
Should we consider doing some measures on several robots and adapt the value mapping on the sliders?

**Comment by motib** [comment]
Hi Fanny,
I have complained of the variation in the horizontal prox sensors and I'm sure many others have too. Stephane send me data from a large number of trials, and the values are widely distributed (my robot had even more extreme values), so it is hopeless to fix thresholds that are good for everyone.

I don't understand why you have to observe the absolute values in Studio. In advanced mode the thresholds can be adjusted by trial-and-error -- a well-known problem-solving technique :-)). The problem only exists in basic mode with its fixed thresholds and I don't think it's too hard to go to advanced mode if someone is having a serious problem in basic mode. That's why I don't complain about this issue any longer!

Please explain what the second line of your program does. The Thymio follows my hand easily when I remove the line.
Moti

PS: Stephane, why is the issue still open? Your original post asked for adjustable threshold and they have been implemented.

**Comment by riedo** [comment]
Maybe I'm just not patient enough with the trial and error :)
I looked at the numerical values because the logic of my code seemed good, but it did not work, and by changing a bit the thresholds I could not really measure the progress. With the feedback from the robot it's easier to compare and know which threshold is too high or too low.

The second line makes the robot stop when the hand is neither too far nor too close; the robot stays quiet and does not oscillate back and forth (like in the green mode). What I did not expect was that I would need to make the "appropriate distance" detection zone so small, while the "too far" detection zone (line 3) is so wide (in reality, in distance, I do not have a very narrow "appropriate distance" and a very large "detected too far" zone)
Does that make sense?

Could I see this data? Unfortunately, I know there is a lot of variation on those sensors, so I expected this answer... I guess it cannot be helped. For me we can close the issue if the mapping cannot be improved.
